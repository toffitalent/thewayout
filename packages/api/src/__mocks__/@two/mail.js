const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
