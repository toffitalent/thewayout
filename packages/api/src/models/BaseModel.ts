import { PaginateQueryBuilder, Serialize } from '@disruptive-labs/objection-plugins';
import { Model, ModelOptions, QueryContext } from 'objection';
import { knex } from '../db';

// Bind knex instance to objection
Model.knex(knex);

export class BaseModel extends Serialize(Model) {
  static QueryBuilder = PaginateQueryBuilder;
  QueryBuilderType!: PaginateQueryBuilder<this>;

  static useLimitInFirst = true;

  createdAt!: Date;
  updatedAt!: Date;

  async $beforeInsert(queryContext: QueryContext) {
    await super.$beforeInsert(queryContext);

    this.createdAt = new Date();
    this.updatedAt = this.createdAt;
  }

  async $beforeUpdate(opt: ModelOptions, queryContext: QueryContext) {
    await super.$beforeUpdate(opt, queryContext);

    this.updatedAt = new Date();
  }
}
