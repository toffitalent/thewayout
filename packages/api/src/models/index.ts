export * from './App';
export * from './AppClient';
export * from './AppToken';
export * from './AppUser';
export * from './BaseModel';
export * from './Client';
export * from './Employer';
export * from './Job';
export * from './JobApplication';
export * from './Rsp';
export * from './RspAccount';
export * from './RspInvitation';
export * from './serializers';
export * from './User';
export {
  CheckViolationError,
  ConstraintViolationError,
  DataError,
  DBError,
  ForeignKeyViolationError,
  NotFoundError,
  NotNullViolationError,
  raw,
  ref,
  UniqueViolationError,
  ValidationError,
} from 'objection';
