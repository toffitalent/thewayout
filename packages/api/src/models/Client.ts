import {
  Age,
  Education,
  Ethnicity,
  ExperienceSkills,
  Gender,
  JusticeStatus,
  Languages,
  License,
  MaritalStatus,
  OffenseCategory,
  Orientation,
  PersonalStrengths,
  ReferredBy,
  RelativeExperience,
  ReleasedAt,
  Religion,
  State,
  StateFederal,
  Support,
  TimeServed,
  VeteranBranchOfService,
  VeteranCampaign,
  VeteranOrJustice,
  VeteranRank,
  VeteranStatus,
  VeteranTypeDischarge,
} from '@two/shared';
import { BaseModel } from './BaseModel';
import { ClientSerializer } from './serializers';

export class Client extends BaseModel {
  static tableName = 'clients';
  static Serializer = ClientSerializer;

  readonly id!: string;

  userId!: string;
  firstName!: string;
  lastName!: string;
  justiceStatus!: JusticeStatus;
  facility!: string | null;
  expectedReleasedAt!: string | null;
  releasedCounty!: string | null;
  phone!: string | null;
  address!: string;
  city!: string;
  state!: State | string;
  postalCode!: string;
  support!: Support[];
  gender!: Gender | null;
  orientation!: Orientation | null;
  religion!: Religion | null;
  maritalStatus!: MaritalStatus | null;
  age!: Age | null;
  disability!: boolean | null;
  ethnicity!: Ethnicity | null;
  veteranStatus!: VeteranStatus | null;
  referredBy!: ReferredBy;
  personalStrengths!: PersonalStrengths[];
  experience!: ExperienceSkills[];
  languages!: Languages[];
  offense!: OffenseCategory[] | null;
  sexualOffenderRegistry!: boolean | null;
  sbn!: boolean | null;
  timeServed!: TimeServed | null;
  releasedAt!: ReleasedAt | null;
  stateOrFederal!: StateFederal | null;
  relativeExperience!: RelativeExperience[];
  personalSummary!: string;
  education!: Education[];
  license!: License[];
  veteranOrJustice!: VeteranOrJustice[];
  veteranService?: VeteranBranchOfService[];
  veteranRank?: VeteranRank;
  veteranReservist?: boolean;
  veteranCampaigns?: VeteranCampaign[];
  veteranStartAt?: string;
  veteranEndAt?: string;
  veteranTypeDischarge?: VeteranTypeDischarge;
  veteranDd214?: boolean;
}
