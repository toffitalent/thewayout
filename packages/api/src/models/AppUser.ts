import type { RelationMappingsThunk } from 'objection';
import { App } from './App';
import { BaseModel } from './BaseModel';
import { AppUserSerializer } from './serializers';
import { User } from './User';

export class AppUser extends BaseModel {
  static tableName = 'app_users';
  static Serializer = AppUserSerializer;

  readonly id!: string;
  appId!: string;
  userId!: string;
  scope!: string[];

  app?: App;
  user?: User;

  static relationMappings: RelationMappingsThunk = () => ({
    app: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: App,
      join: {
        from: 'app_users.appId',
        to: 'apps.id',
      },
    },

    user: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: User,
      join: {
        from: 'app_users.userId',
        to: 'users.id',
      },
    },
  });
}
