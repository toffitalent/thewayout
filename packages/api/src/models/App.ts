import type { RelationMappingsThunk } from 'objection';
import { AppClient } from './AppClient';
import { AppToken } from './AppToken';
import { AppUser } from './AppUser';
import { BaseModel } from './BaseModel';
import { AppSerializer } from './serializers';

export class App extends BaseModel {
  static tableName = 'apps';
  static Serializer = AppSerializer;

  readonly id!: string;
  name!: string;
  userId!: string | null;

  clients?: AppClient[];
  tokens?: AppToken[];
  users?: AppUser[];

  static relationMappings: RelationMappingsThunk = () => ({
    clients: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppClient,
      join: {
        from: 'apps.id',
        to: 'app_clients.appId',
      },
    },

    tokens: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppToken,
      join: {
        from: 'apps.id',
        to: 'app_tokens.appId',
      },
    },

    users: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppUser,
      join: {
        from: 'apps.id',
        to: 'app_users.appId',
      },
    },
  });
}
