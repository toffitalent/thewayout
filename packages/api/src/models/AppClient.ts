import type { RelationMappingsThunk } from 'objection';
import { AuthGrantType } from '@two/shared';
import { App } from './App';
import { AppToken } from './AppToken';
import { BaseModel } from './BaseModel';

export class AppClient extends BaseModel {
  static tableName = 'app_clients';

  readonly id!: string;
  appId!: string;
  name!: string;
  secret!: string;
  grantType!: AuthGrantType;
  redirectUri!: string | null;

  app?: App;
  tokens?: AppToken[];

  static relationMappings: RelationMappingsThunk = () => ({
    app: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: App,
      join: {
        from: 'app_clients.appId',
        to: 'apps.id',
      },
    },

    tokens: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppToken,
      join: {
        from: 'app_clients.id',
        to: 'app_tokens.clientId',
      },
    },
  });
}
