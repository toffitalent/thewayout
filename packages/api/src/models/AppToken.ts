import type { RelationMappingsThunk } from 'objection';
import { App } from './App';
import { AppClient } from './AppClient';
import { BaseModel } from './BaseModel';
import { User } from './User';

export class AppToken extends BaseModel {
  static tableName = 'app_tokens';

  readonly id!: string;
  appId!: string;
  clientId!: string;
  userId!: string;
  token!: Buffer;
  self!: boolean;

  app?: App;
  client?: AppClient;
  user?: User;

  static relationMappings: RelationMappingsThunk = () => ({
    app: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: App,
      join: {
        from: 'app_tokens.appId',
        to: 'apps.id',
      },
    },

    client: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: AppClient,
      join: {
        from: 'app_tokens.clientId',
        to: 'app_clients.id',
      },
    },

    user: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: User,
      join: {
        from: 'app_tokens.userId',
        to: 'users.id',
      },
    },
  });
}
