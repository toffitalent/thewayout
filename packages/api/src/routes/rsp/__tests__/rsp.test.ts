import { faker } from '@faker-js/faker';
import { send } from '@two/mail';
import { RspAccount, RspInvitation, User } from '@app/models';
import { auth, fixtures, request, resetDb } from '@test';

const { rsp1, user7, user8, rspAccount2 } = fixtures;

describe('Routes > Rsp > Rsp', () => {
  beforeEach(async () => {
    await resetDb();

    faker.seed(1);
  });

  describe('PATCH /rsp/:id', () => {
    test('updates rsp organization', async () => {
      const res = await request()
        .patch(`/api/v1/rsp/${rsp1.id}`)
        .send({ name: 'Test' })
        .use(auth(user7.id, ['rsp']));

      expect(res.status).toBe(200);
      expect(res.body.name).toBe('Test');
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });

    test('updates rsp organization only by owner', async () => {
      const res = await request()
        .patch(`/api/v1/rsp/${rsp1.id}`)
        .send({ name: 'Test' })
        .use(auth(user8.id, ['rsp']));

      expect(res.status).toBe(400);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('POST /rsp/:id/accounts/invite', () => {
    test('sends invitation', async () => {
      const res = await request()
        .post(`/api/v1/rsp/${rsp1.id}/accounts/invite`)
        .use(auth(user7.id, ['rsp']))
        .send({
          firstName: 'First',
          lastName: 'Last',
          phone: '4142133945',
          email: 'email@test.com',
        });

      const invitation = await RspInvitation.query()
        .where({ email: 'email@test.com' })
        .first()
        .throwIfNotFound();

      expect(send).toBeCalledTimes(1);
      expect((send as jest.Mock).mock.calls[0][0]).toEqual({
        to: {
          address: 'email@test.com',
          name: 'First Last',
        },
      });
      expect((send as jest.Mock).mock.calls[0][1]).toBe('InviteCaseManager');
      expect((send as jest.Mock).mock.calls[0][2]).toStrictEqual({
        firstName: 'First',
        lastName: 'Last',
        phone: '4142133945',
        email: 'email@test.com',
        invitationId: invitation.id,
        rspName: rsp1.name,
      });

      expect(res.status).toBe(204);
    });

    test('only owner can sends invitation', async () => {
      const res = await request()
        .post(`/api/v1/rsp/${rsp1.id}/accounts/invite`)
        .use(auth(user8.id, ['rsp']))
        .send({
          firstName: 'First',
          lastName: 'Last',
          phone: '4142133945',
          email: 'email@test.com',
        });

      expect(send).toBeCalledTimes(0);

      expect(res.status).toBe(400);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });

    test('returns a resource conflict response for duplicate invitation', async () => {
      const email = faker.internet.email().toLowerCase();
      await RspInvitation.query().insert({
        rspId: fixtures.rsp1.id,
        email,
      });

      const res = await request()
        .post(`/api/v1/rsp/${rsp1.id}/accounts/invite`)
        .use(auth(user7.id, ['rsp']))
        .send({
          firstName: 'First',
          lastName: 'Last',
          phone: '4142133945',
          email,
        });

      expect(res.status).toBe(409);
    });

    test('returns a invalid request response for duplicate email addresses', async () => {
      const res = await request()
        .post(`/api/v1/rsp/${rsp1.id}/accounts/invite`)
        .use(auth(user7.id, ['rsp']))
        .send({
          firstName: 'First',
          lastName: 'Last',
          phone: '4142133945',
          email: user8.email,
        });

      expect(res.status).toBe(409);
    });
  });

  describe('GET /rsp/:id/invitations', () => {
    test('lists invitations', async () => {
      await RspInvitation.query().insert({
        rspId: fixtures.rsp1.id,
        email: faker.internet.email().toLowerCase(),
      });

      const res = await request()
        .get(`/api/v1/rsp/${rsp1.id}/invitations`)
        .use(auth(user7.id, ['rsp']));

      expect(res.status).toBe(200);
      expect(res.type).toBe('application/json');
      expect({ body: res.body }).toMatchSnapshot({ body: [{ id: expect.any(String) }] });
    });

    test('lists only pending invitations', async () => {
      const email = faker.internet.email().toLowerCase();
      await RspInvitation.query().insert({
        rspId: fixtures.rsp1.id,
        email,
      });
      const newUser = await User.query().insert({ email });
      await RspAccount.query().insert({
        userId: newUser.id,
        rspId: fixtures.rsp1.id,
        isProfileFilled: true,
      });

      const res = await request()
        .get(`/api/v1/rsp/${rsp1.id}/invitations`)
        .use(auth(user7.id, ['rsp']));

      expect(res.status).toBe(200);
      expect(res.type).toBe('application/json');
      expect(res.body.length).toBe(0);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('DELETE /rsp/:id/invitations/:invitationId', () => {
    test('removes invitation', async () => {
      const invitation = await RspInvitation.query().insert({
        rspId: fixtures.rsp1.id,
        email: faker.internet.email().toLowerCase(),
      });

      const res = await request()
        .delete(`/api/v1/rsp/${rsp1.id}/invitations/${invitation.id}`)
        .use(auth(user7.id, ['rsp']));

      const removedInvitation = await RspInvitation.query().findById(invitation.id);
      expect(removedInvitation).toBeUndefined();

      expect(res.status).toBe(204);
    });

    test('removes invitation only for owner', async () => {
      const invitation = await RspInvitation.query().insert({
        rspId: fixtures.rsp1.id,
        email: faker.internet.email().toLowerCase(),
      });

      const res = await request()
        .delete(`/api/v1/rsp/${rsp1.id}/invitations/${invitation.id}`)
        .use(auth(user8.id, ['rsp']));

      expect(res.status).toBe(400);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('GET /rsp/:id/case-managers', () => {
    test('lists case managers', async () => {
      const res = await request()
        .get(`/api/v1/rsp/${rsp1.id}/case-managers`)
        .use(auth(user7.id, ['rsp']));

      expect(res.status).toBe(200);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });

    test('lists case managers only for owner', async () => {
      const res = await request()
        .get(`/api/v1/rsp/${rsp1.id}/case-managers`)
        .use(auth(user8.id, ['rsp']));

      expect(res.status).toBe(400);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('PATCH /rsp/:id/accounts/:id', () => {
    test('updates case manager account', async () => {
      expect(user8.firstName).not.toBe('New');
      const res = await request()
        .patch(`/api/v1/rsp/${rsp1.id}/accounts/${rspAccount2.userId}`)
        .send({ firstName: 'New' })
        .use(auth(user7.id, ['rsp']));

      expect(res.status).toBe(200);
      expect(res.body.firstName).toBe('New');
    });

    test('updates case manager account only by owner', async () => {
      const res = await request()
        .patch(`/api/v1/rsp/${rsp1.id}/accounts/${rspAccount2.userId}`)
        .send({ firstName: 'New' })
        .use(auth(user8.id, ['rsp']));

      expect(res.status).toBe(400);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('GET /rsp/:id/accounts/:userId', () => {
    test('returns rsp account', async () => {
      const res = await request()
        .get(`/api/v1/rsp/${rsp1.id}/accounts/${rspAccount2.userId}`)
        .use(auth(user7.id, ['rsp']));

      expect(res.status).toBe(200);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });

    test('returns rsp account only for owner', async () => {
      const res = await request()
        .get(`/api/v1/rsp/${rsp1.id}/accounts/${rspAccount2.userId}`)
        .use(auth(user8.id, ['rsp']));

      expect(res.status).toBe(400);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('DELETE /rsp/:id/accounts/:userId', () => {
    test('removes rsp account', async () => {
      const res = await request()
        .delete(`/api/v1/rsp/${rsp1.id}/accounts/${rspAccount2.userId}`)
        .use(auth(user7.id, ['rsp']));

      expect(res.status).toBe(204);
    });

    test('removes rsp account only for owner', async () => {
      const res = await request()
        .delete(`/api/v1/rsp/${rsp1.id}/accounts/${rspAccount2.userId}`)
        .use(auth(user8.id, ['rsp']));

      expect(res.status).toBe(400);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });
  });
});
