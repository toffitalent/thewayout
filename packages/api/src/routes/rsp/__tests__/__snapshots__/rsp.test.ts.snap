// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes > Rsp > Rsp DELETE /rsp/:id/accounts/:userId removes rsp account only for owner 1`] = `
{
  "code": "InvalidRequest",
  "errors": [
    {
      "error": "invalid",
      "key": [
        "owner",
      ],
      "source": "body",
    },
  ],
  "message": "Allowed only to owners",
  "status": 400,
}
`;

exports[`Routes > Rsp > Rsp DELETE /rsp/:id/invitations/:invitationId removes invitation only for owner 1`] = `
{
  "code": "InvalidRequest",
  "errors": [
    {
      "error": "invalid",
      "key": [
        "owner",
      ],
      "source": "body",
    },
  ],
  "message": "Allowed only to owners",
  "status": 400,
}
`;

exports[`Routes > Rsp > Rsp GET /rsp/:id/accounts/:userId returns rsp account 1`] = `
{
  "avatar": null,
  "email": "mariah_barton@example.com",
  "firstName": "Mariah",
  "id": "01817e24-0d42-786a-a454-7291c0173657",
  "isProfileFilled": true,
  "lastName": "Barton",
  "phone": "1231231234",
  "role": "member",
  "rsp": {
    "address": "Ibrahim Pine",
    "avatar": null,
    "city": "East Barneybury",
    "createdAt": "<Date>",
    "description": "Enim sequi iusto.
Explicabo quisquam omnis nisi corporis corrupti.
Cupiditate qui culpa numquam iure quisquam vero architecto.",
    "email": null,
    "id": "01817e24-0d42-77e5-9a94-ad514d8ec5dc",
    "justiceStatus": [
      "halfwayHouse",
      "parole",
    ],
    "name": "Oberbrunner and Sons",
    "offenses": [
      "burglary",
      "distribution",
      "diuDwi",
      "embezzlement",
    ],
    "phone": null,
    "postalCode": "78580",
    "servicesArea": [
      "Adams",
      "Ashland",
      "Barron",
    ],
    "state": "AK",
    "support": [
      "basicHygieneSupport",
      "clothing",
      "basicTechSkills",
      "childCare",
    ],
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "justiceImpacted",
    ],
  },
  "rspId": "01817e24-0d42-77e5-9a94-ad514d8ec5dc",
  "userId": "01817e24-0d42-75e5-8c38-8eb33c91884c",
}
`;

exports[`Routes > Rsp > Rsp GET /rsp/:id/accounts/:userId returns rsp account only for owner 1`] = `
{
  "code": "InvalidRequest",
  "errors": [
    {
      "error": "invalid",
      "key": [
        "owner",
      ],
      "source": "body",
    },
  ],
  "message": "Allowed only to owners",
  "status": 400,
}
`;

exports[`Routes > Rsp > Rsp GET /rsp/:id/case-managers lists case managers 1`] = `
[
  {
    "avatar": null,
    "caseLoad": 26,
    "email": "mariah_barton@example.com",
    "firstName": "Mariah",
    "id": "01817e24-0d42-786a-a454-7291c0173657",
    "lastName": "Barton",
    "phone": "1231231234",
    "userId": "01817e24-0d42-75e5-8c38-8eb33c91884c",
  },
  {
    "avatar": null,
    "caseLoad": 0,
    "email": "quinten.mcdermott@example.com",
    "firstName": "Quinten",
    "id": "01817e24-0d42-7846-9a0b-4a6f781f0513",
    "lastName": "McDermott",
    "phone": "1231231234",
    "userId": "01817e24-0d42-75da-8f84-bfaed6311cda",
  },
]
`;

exports[`Routes > Rsp > Rsp GET /rsp/:id/case-managers lists case managers only for owner 1`] = `
{
  "code": "InvalidRequest",
  "errors": [
    {
      "error": "invalid",
      "key": [
        "owner",
      ],
      "source": "body",
    },
  ],
  "message": "Allowed only to owners",
  "status": 400,
}
`;

exports[`Routes > Rsp > Rsp GET /rsp/:id/invitations lists invitations 1`] = `
{
  "body": [
    {
      "email": "zion.reichel12@yahoo.com",
      "firstName": null,
      "id": Any<String>,
      "lastName": null,
      "phone": null,
      "rspId": "01817e24-0d42-77e5-9a94-ad514d8ec5dc",
    },
  ],
}
`;

exports[`Routes > Rsp > Rsp GET /rsp/:id/invitations lists only pending invitations 1`] = `[]`;

exports[`Routes > Rsp > Rsp PATCH /rsp/:id updates rsp organization 1`] = `
{
  "address": "Ibrahim Pine",
  "avatar": null,
  "city": "East Barneybury",
  "createdAt": "<Date>",
  "description": "Enim sequi iusto.
Explicabo quisquam omnis nisi corporis corrupti.
Cupiditate qui culpa numquam iure quisquam vero architecto.",
  "email": null,
  "id": "01817e24-0d42-77e5-9a94-ad514d8ec5dc",
  "justiceStatus": [
    "halfwayHouse",
    "parole",
  ],
  "name": "Test",
  "offenses": [
    "burglary",
    "distribution",
    "diuDwi",
    "embezzlement",
  ],
  "phone": null,
  "postalCode": "78580",
  "servicesArea": [
    "Adams",
    "Ashland",
    "Barron",
  ],
  "state": "AK",
  "support": [
    "basicHygieneSupport",
    "clothing",
    "basicTechSkills",
    "childCare",
  ],
  "updatedAt": "<Date>",
  "veteranOrJustice": [
    "justiceImpacted",
  ],
}
`;

exports[`Routes > Rsp > Rsp PATCH /rsp/:id updates rsp organization only by owner 1`] = `
{
  "code": "InvalidRequest",
  "errors": [
    {
      "error": "invalid",
      "key": [
        "owner",
      ],
      "source": "body",
    },
  ],
  "message": "Allowed only to owners",
  "status": 400,
}
`;

exports[`Routes > Rsp > Rsp PATCH /rsp/:id/accounts/:id updates case manager account only by owner 1`] = `
{
  "code": "InvalidRequest",
  "errors": [
    {
      "error": "invalid",
      "key": [
        "owner",
      ],
      "source": "body",
    },
  ],
  "message": "Allowed only to owners",
  "status": 400,
}
`;

exports[`Routes > Rsp > Rsp POST /rsp/:id/accounts/invite only owner can sends invitation 1`] = `
{
  "code": "InvalidRequest",
  "errors": [
    {
      "error": "invalid",
      "key": [
        "owner",
      ],
      "source": "body",
    },
  ],
  "message": "Allowed only to owners",
  "status": 400,
}
`;
